File: CPP/String/D17_NonRepeatingCharacter.cpp
#include <iostream>
#include <string>

using namespace std;

char nonRepeatingChar(string& s) {
      int freq[26];
      std::fill(freq, freq + 26, 0); 

      for (auto ch : s) {
            freq[ch - 'a'] ++;
      }

      for (auto ch : s) {
            if (freq[ch - 'a'] == 1) {
                  return ch;
            }
      }
      
      return '$';
}

int main() {
      string s1 = "geeksforgeeks", s2 = "racecar", s3 = "aabbccc";
      cout << nonRepeatingChar(s1) << endl ;
      cout << nonRepeatingChar(s2) << endl ;
      cout << nonRepeatingChar(s3) << endl ;
      return 0;
}

File: Python/String/D17_NonRepeatingCharacter.py
def non_repeating_char(s):
      freq = [0] * 26

      for c in s:
            freq[ ord(c) - ord('a') ] += 1

      for c in s:
            if freq[ord(c) - ord('a') ] == 1:
                  return c
            
      return '$'

print(non_repeating_char("geeksforgeeks")) 
print(non_repeating_char("racecar"))
print(non_repeating_char("aabbcc"))